public/js/NeighbourPlot.js

```diff
@@ -0,0 +1,199 @@
+var d3 = require('d3');
+require('d3-tip')(d3);
+
+/**
+ * Draws PCA scatterplot of the data Clicking on scatterplot points will
```

Comment:
How about adding the function name here? Man Python really got it right with their docstrings, they should definitely come *after* the function name! I started reading this commend going huh? wtf is he talking about? Then I scrolled down and realised it was a JSDoc! How about:

```
/**
 * NeighbourPlot: class to draw scatterplot of dimension-reduced data
 *
 * Clicking on scatterplot points will trigger updates of the neighbour image
 * [...]
```

What do you think?